libs/ngrx-toolkit/src/lib/with-resource.spec.ts

```diff
@@ -0,0 +1,662 @@
+            expect(store.addressValue()).toBeUndefined();
+            expect(store.addressError()).toBeUndefined();
+            expect(store.addressIsLoading()).toBe(true);
+            expect(store.addressHasValue()).toBe(false);
+          });
+
+          it('has resolved status when loaded', async () => {
+            const { store, addressResolver } = setupWithNamedResource();
+
+            addressResolver.resolve.mockResolvedValue(venice);
+            patchState(store, { id: 1 });
+
+            await wait();
+
+            expect(store.addressStatus()).toBe('resolved');
+            expect(store.addressValue()).toEqual(venice);
+            expect(store.addressError()).toBeUndefined();
+            expect(store.addressIsLoading()).toBe(false);
+            expect(store.addressHasValue()).toBe(true);
+          });
+
+          it('has error status when error', async () => {
+            const { store, addressResolver } = setupWithNamedResource();
+
+            addressResolver.resolve.mockRejectedValue(new Error('Error'));
+            patchState(store, { id: 1 });
+            await wait();
+
+            expect(store.addressStatus()).toBe('error');
+            expect(() => store.addressValue()).toThrow();
+            expect(store.addressError()).toBeInstanceOf(Error);
+            expect(store.addressIsLoading()).toBe(false);
+            expect(store.addressHasValue()).toBe(false);
+          });
+
+          it('has local once updated', async () => {
+            const { store, addressResolver } = setupWithNamedResource();
+
+            addressResolver.resolve.mockResolvedValue(venice);
+            patchState(store, { id: 1 });
+
+            await wait();
+            patchState(store, ({ addressValue }) => ({
+              addressValue: addressValue
+                ? { ...addressValue, country: 'Italia' }
+                : undefined,
+            }));
+
+            expect(store.addressStatus()).toBe('local');
+            expect(store.addressValue()?.country).toBe('Italia');
+            expect(store.addressError()).toBeUndefined();
+            expect(store.addressIsLoading()).toBe(false);
+            expect(store.addressHasValue()).toBe(true);
+          });
+
+          it('can also reload by resource name', async () => {
+            const { store, addressResolver } = setupWithNamedResource();
+
+            addressResolver.resolve.mockResolvedValueOnce(venice);
+            patchState(store, { id: 1 });
+            await wait();
+            expect(store.addressStatus()).toBe('resolved');
+            store.reload();
+            expect(store.addressStatus()).toBe('reloading');
+          });
+        });
+      });
+
+      describe('override protection', () => {
+        const warningSpy = jest.spyOn(console, 'warn');
+
+        afterEach(() => {
+          warningSpy.mockClear();
+        });
+
+        for (const memberName of [
+          //TODO wait for https://github.com/ngrx/platform/pull/4932
```

Comment:
Will this PR waiting for this to land in the store first? Or is this just a note to keep in mind that it will eventually be ready.

Reply:
No, it is really just a note for us. We will release earlier